google/cloud/storage/parallel_uploads_test.cc

```diff
@@ -150,178 +149,130 @@ class ParallelUploadTest
+                            &UploadChunkRequest::upload_session_url, id)))
+        .WillOnce([=](UploadChunkRequest const& r) {
+          if (expected_content) {
+            EXPECT_THAT(r.payload(),
+                        ElementsAre(ConstBuffer(*expected_content)));
+          }
+          return QueryResumableUploadResponse{
+              r.offset() + r.payload_size(),
+              MockObject(object_name, generation)};
+        });
     return id;
   }
 
   std::string ExpectResumeSession(
       std::string const& object_name, int generation, std::string id,
       absl::optional<std::uint64_t> const& committed_size = absl::nullopt,
       absl::optional<std::string> const& expected_content = absl::nullopt) {
-    auto saved = SaveSessionId(id);
-    EXPECT_CALL(*mock_, CreateResumableSession(Property(
-                            &ResumableUploadRequest::object_name, object_name)))
-        .WillOnce([=](internal::ResumableUploadRequest const& request) {
-          EXPECT_THAT(
-              request.GetOption<UseResumableUploadSession>().value_or(""), id);
-          EXPECT_EQ(kBucketName, request.bucket_name());
-
-          auto session =
-              absl::make_unique<testing::MockResumableUploadSession>();
-          EXPECT_CALL(*session, session_id).WillRepeatedly(ReturnRef(*saved));
-          EXPECT_CALL(*session, UploadFinalChunk)
-              .WillRepeatedly([=](ConstBufferSequence const& content,
-                                  std::uint64_t size,
-                                  HashValues const& /*full_object_hashes*/) {
-                if (expected_content) {
-                  EXPECT_THAT(content,
-                              ElementsAre(ConstBuffer(*expected_content)));
-                }
-                return ResumableUploadResponse{
-                    id, ResumableUploadResponse::kDone, size,
-                    MockObject(object_name, generation),
-                    /*annotations=*/std::string{}};
-              });
-          return CreateResumableSessionResponse{
-              std::move(session),
-              ResumableUploadResponse{id, ResumableUploadResponse::kInProgress,
-                                      /*.committed_size=*/committed_size,
-                                      /*.object_metadata=*/absl::nullopt,
-                                      /*.annotations=*/std::string{}}};
+    EXPECT_CALL(*mock_,
+                QueryResumableUpload(Property(
+                    &QueryResumableUploadRequest::upload_session_url, id)))
+        .WillOnce([=](internal::QueryResumableUploadRequest const&) {
+          return internal::QueryResumableUploadResponse{committed_size,
+                                                        absl::nullopt};
         })
         .RetiresOnSaturation();
+
+    using ::testing::AtMost;
+    EXPECT_CALL(*mock_, UploadChunk(Property(
+                            &UploadChunkRequest::upload_session_url, id)))
+        .Times(AtMost(1))
```

Comment:
I guess this isn't a `WillOnce(...)` case?

For some reason, I thought `.Times(0).WillRepeatedly(...)` didn't work. Which makes me think that `.Times(AtMost(1))` is really `.Times(1)`.... but idk

Reply:
Somehow I had convinced myself that this could be called one or zero times.  Changed to `.WillOnce()`.